src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -920,21 +957,20 @@ private List<PatternDescription> reduceBindingPatterns(Type selectorType, List<P
                                     PERMITTED: for (Iterator<Symbol> it = permitted.iterator(); it.hasNext();) {
                                         Symbol perm = it.next();
 
-                                        if (types.isSubtype(types.erasure(perm.type),
-                                                            types.erasure(bpOther.type))) {
-                                            it.remove();
-                                            reduces = true;
-                                            continue PERMITTED;
-                                        }
-
                                         for (Symbol currentPermitted : currentPermittedSubTypes) {
                                             if (types.isSubtype(types.erasure(currentPermitted.type),
                                                                 types.erasure(perm.type))) {
                                                 it.remove();
-                                                reduces = true;
                                                 continue PERMITTED;
                                             }
                                         }
+                                        if (types.isSubtype(types.erasure(perm.type),
+                                                            types.erasure(bpOther.type))) {
+                                            it.remove();
+                                            reduces = true;
+                                            continue ;
```

Comment:
This continue seems redundant

Reply:
Thanks, fixed:
https://github.com/openjdk/jdk/pull/13074/commits/8b1cfba6e4603d82836d889273855185165acd29